jenkins/bump-clio-version.sh

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+
+# Turn on verbose
+set -ex
+
+CALLING_DIR=$(pwd)
+CLIO_DIR=$(cd $(dirname $(dirname $0)) && pwd)
+
+# Run this script on Jenkins as part of the stage-RC process to
+# create and commit a bump to Clio's base version.
+
+# Expects the env var RELEASED_VERSION to be set to the full value of the just-released version.
+
+# If run on develop, bumps the minor version.
+# Otherwise (for a hotfix branch) bumps the patch version.
+
+if [ -z "$RELEASED_VERSION" ]
+then
+  >&2 echo "Error: RELEASED_VERSION not set!!"
+  exit 1
+fi
+
+VERSION_FILE="${CLIO_DIR}/.clio-version"
+if [ ! -f "$VERSION_FILE" ]
+then
+  >&2 echo "Error: No version file at '${VERSION_FILE}'!!"
+  exit 1
+fi
+
+cd "$CLIO_DIR"
+COMMIT=$(git rev-parse HEAD)
+BASE_VERSION=$(cat "${CLIO_DIR}/.clio-version" | tr -d '\n')
+
+# First, tag the current commit with the just-released version.
+git tag "$RELEASED_VERSION"
```

Comment:
For all these pushes I was thinking it wouldn't be possible because of the SSH key mismatch, but this runs in a separate job so I can probably configure it to use the GitHub SSH key for everything.